db/sqlinterfaces.c

```diff
@@ -4629,8 +4629,11 @@ int check_active_appsock_connections(struct sqlclntstate *clnt)
         int num_retry = 0;
         int rc = -1;
     retry:
+        /* Don't count this connection until we are done with the check */
+        active_appsock_conns--;
```

Comment:
Don't we need appsock_conn_lk here to modify active_appsock_conns? We can also go over the limit if the dirty check fails for a bunch of connections all at once.

Reply:
yes. we have another check and then retry.